misk/src/main/kotlin/misk/digester/SlidingWindowDigest.kt

```diff
@@ -0,0 +1,188 @@
+package misk.digester
+
+import java.time.Clock
+import java.time.ZonedDateTime
+
+/** WindowDigest holds a t-digest whose data points are scoped to a specific time window. */
+data class WindowDigest(
+  var window: Window,
+  var Digest: TDigest
+)
+
+/** Snapshot is the state of a SlidingWindowDigest at a point in time. */
+data class Snapshot(
+  var quantileVals: DoubleArray,  // Values of specific quantiles.
+  var count: Long,                // Count of observations.
+  var sum: Double                 // Sum of observations.
+)
+
+/**
+ * SlidingWindowDigest approximates quantiles of data for a trailing time period. It is thread-safe.
+ *
+ * To efficiently store observed data in a mergeable way, t-openDigests are used.
+ * As t-openDigests do not support discarding older data points, the sliding window
+ * aspect is approximated by keeping multiple separate t-openDigests scoped to discrete
+ * overlapping time windows. As a result, quantile data is reported from the most recent window that has ended.
+ *
+ * The following example creates a 1 minute sliding window where there are 6 overlapping windows at a given time.
+ * Reported quantiles are at most 10 seconds out of date.
+ * NewSlidingWindowDigest(time.Now, NewDefaultVeneurDigest, NewWindower(60, 6))
+ *
+ * The following example creates a 10 second sliding window where there are 2 overlapping windows at a given time.
+ * Reported quantiles are at most 5 seconds out of date:
+ * NewSlidingWindowDigest(time.Now, NewDefaultVeneurDigest, NewWindower(10, 2))
+ */
+class SlidingWindowDigest constructor(
+  internal var utcNowClock: Clock,
+  internal val windower: Windower
+) {
+
+  internal var windows: MutableList<WindowDigest> = mutableListOf()
```

Comment:
Why internal? Keep as much private as possible.

Reply:
Testcases that need windows cannot access it when private